crates/cubecl-std/src/reduce/sum.rs

```diff
@@ -0,0 +1,56 @@
+use cubecl_core as cubecl;
+use cubecl_core::prelude::*;
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+pub struct ReduceConfig {
+    pub line_size: u32,
+    pub plane_size: u32,
+    pub num_planes: u32,
+}
+
+/// Compute the sum of all elements of `input` and write it to the first element of `output`.
+///
+/// This is a work in progress toward a more general multi-dimensional reduce kernel.
+#[cube(launch_unchecked)]
+pub fn reduce_sum<N: Numeric>(
+    input: &Tensor<Line<N>>,
+    output: &mut Tensor<Line<N>>,
+    #[comptime] config: ReduceConfig,
+) {
+    reduce_sum_vector(&input.to_slice(), &mut output.to_slice_mut(), config);
+}
+
+#[cube]
+fn reduce_sum_vector<N: Numeric>(
+    input: &Slice<Line<N>>,
+    output: &mut SliceMut<Line<N>>,
+    #[comptime] config: ReduceConfig,
+) {
+    let block_size = config.plane_size * config.num_planes;
+
+    // This is an integer division rounded up.
+    let num_blocks = input.len() / block_size + (input.len() % block_size > 0) as u32;
+
+    let mut memory = SharedMemory::new_lined(config.plane_size, config.line_size);
+    memory[UNIT_POS_X] = Line::empty(config.line_size).fill(N::from_int(0));
+
+    // For each block, we reduce each plane to a single value. Then, we accumulate the results
+    // into the memory. Thus, after the loop, the reduction of the first num_planes values
+    // of the memory yields the expected output.
+    for i in 0..num_blocks {
+        let start = i * block_size + UNIT_POS_Y * config.plane_size;
+        let sum = plane_sum(input[start + UNIT_POS_X]);
```

Comment:
When only reduce tests are run, they all pass, but when run with all other tests, some fail:
```
---- tests::reduce_sum_vector_f32_too_many_planes stdout ----
thread 'tests::reduce_sum_vector_f32_too_many_planes' panicked at /Users/louisfortier-dubois/Projects/cubecl/crates/cubecl-std/src/reduce/test.rs:268:9:
Values differ more than epsilon: index=0 actual=8124.687, expected=8128, difference=3.3129883, epsilon=0.000000001
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- tests::reduce_sum_vector_with_remainder_four_planes stdout ----
thread 'tests::reduce_sum_vector_with_remainder_four_planes' panicked at /Users/louisfortier-dubois/Projects/cubecl/crates/cubecl-std/src/reduce/test.rs:259:17:
assertion `left == right` failed
  left: [1517576986]
 right: [4950]

---- tests::reduce_sum_vector_with_remainder_single_plane stdout ----
thread 'tests::reduce_sum_vector_with_remainder_single_plane' panicked at /Users/louisfortier-dubois/Projects/cubecl/crates/cubecl-std/src/reduce/test.rs:259:17:
assertion `left == right` failed
  left: [1841255236]
 right: [4950]
 ```
 
 This is typically due to out of bounds. I think if you make sure start + UNIT_POS_X < input.len() it will be fine

Reply:
What command did you ran to get these failures ?

Reply:
Just `cargo test` inside of `crates/cubecl-wgpu`